src/app/parcels/parcelsTable/ParcelsPage.tsx

```diff
@@ -75,22 +76,23 @@ const ParcelsPage: React.FC<{}> = () => {
 
     const [errorMessage, setErrorMessage] = useState<string | null>(null);
 
+    const [modalErrorMessage, setModalErrorMessage] = useState<string | null>(null);
+
     const parcelsTableFetchAbortController = useRef<AbortController | null>(null);
 
     const selectedParcelMessage = getSelectedParcelCountMessage(checkedParcelIds.length);
 
-    const fetchAndSetClientIdForSelectedParcel = useCallback((): void => {
+    const fetchAndSetClientIdForSelectedParcel = useCallback(async (): Promise<void> => {
         if (parcelId === null) {
             return;
         }
 
-        getClientIdForSelectedParcel(parcelId)
-            .then((clientId) => setClientIdForSelectedParcel(clientId))
-            .catch((error) => {
-                if (error instanceof Error) {
-                    setErrorMessage(error.message);
-                }
-            });
+        const { clientId, error } = await getClientIdForSelectedParcel(parcelId);
+        if (error) {
+            setModalErrorMessage(getClientIdErrorMessage(error));
```

Comment:
we need to display this error message in the modal otherwise it's not visible until the modal is closed